editor/animation_track_editor.cpp

```diff
@@ -6714,6 +6762,142 @@ void AnimationTrackEditor::_edit_menu_pressed(int p_option) {
 			AnimationPlayerEditor::get_singleton()->get_player()->apply_reset(true);
 		} break;
 
+		case EDIT_CONVERT_TO_BEZIER: {
+			track_convert_select->clear();
+			TreeItem *troot = track_convert_select->create_item();
+
+			for (int i = 0; i < animation->get_track_count(); i++) {
+				if (animation->track_get_key_count(i) == 0) {
+					continue; // Don't list animations with no keys.
+				}
+				NodePath path = animation->track_get_path(i);
+				Node *node = nullptr;
+
+				if (root) {
+					node = root->get_node_or_null(path);
+				}
+
+				String text;
+				Ref<Texture2D> icon = get_editor_theme_icon(SNAME("Node"));
+				if (node) {
+					if (has_theme_icon(node->get_class(), EditorStringName(EditorIcons))) {
+						icon = get_editor_theme_icon(node->get_class());
+					}
+
+					text = node->get_name();
+					Vector<StringName> sn = path.get_subnames();
+					for (int j = 0; j < sn.size(); j++) {
+						text += ".";
+						text += sn[j];
+					}
+
+					path = NodePath(node->get_path().get_names(), path.get_subnames(), true);
+				} else {
+					text = path;
+					int sep = text.find(":");
+					if (sep != -1) {
+						text = text.substr(sep + 1, text.length());
+					}
+				}
+
+				String track_type;
+				switch (animation->track_get_type(i)) {
+					case Animation::TYPE_VALUE:
+						break; // Show only the node path.
+					case Animation::TYPE_POSITION_3D:
+						track_type = TTR("Position");
+						break;
+					case Animation::TYPE_ROTATION_3D:
+						track_type = TTR("Rotation");
+						break;
+					case Animation::TYPE_SCALE_3D:
+						track_type = TTR("Scale");
+						break;
+					case Animation::TYPE_BLEND_SHAPE:
+						track_type = TTR("BlendShape");
+						break;
+					case Animation::TYPE_METHOD:
+						continue; // Type cannot be converted - skip to next track.
+					case Animation::TYPE_BEZIER:
+						continue; // Type cannot be converted - skip to next track.
+					case Animation::TYPE_AUDIO:
+						continue; // Type cannot be converted - skip to next track.
+					case Animation::TYPE_ANIMATION:
+						continue; // Type cannot be converted - skip to next track.
+				}
```

Comment:
@TokageItLab I wonder if we can add another mode for interpolation?

Reply:
I think it's okay to ignore everything except ValueTrack.